deepgram/clients/prerecorded_client.py

```diff
@@ -0,0 +1,154 @@
+from ..types.prerecorded_response import SyncPrerecordedResponse
+
+class PreRecordedClient(AbstractRestfulClient):
+    """
+    A client class for handling pre-recorded audio data. Provides methods for transcribing audio from URLs and files.
+    """
+    def __init__(self, url, headers):
+        """
+        Initializes a new instance of the PreRecordedClient.
+
+        Args:
+            url (str): The URL for API requests.
+            headers (dict): Headers to include in API requests.
+        """
+        self.url = url
+        self.headers = headers
+        super().__init__(url, headers)
+    
+    async def transcribe_url(
+        self, source: UrlSource, options: PrerecordedOptions = None, endpoint: str="v1/listen"
+    ) -> SyncPrerecordedResponse:
+        """
+        Transcribes audio from a URL source.
+
+        Args:
+            source (UrlSource): The URL source of the audio to transcribe.
+            options (PrerecordedOptions): Additional options for the transcription (default is None).
+            endpoint (str): The API endpoint for the transcription (default is "v1/listen").
+
+        Returns:
+            SyncPrerecordedResponse: An object containing the transcription result.
+
+        Raises:
+            DeepgramError: If the "callback" option is provided for a synchronous transcription.
+            DeepgramApiError: Raised for known API errors.
+            DeepgramUnknownApiError: Raised for unknown API errors.
+            DeepgramUnknownError: Raised for unexpected errors not specific to the API.
+            Exception: For any other unexpected exceptions.
+        """
+
+        url = f"{self.url}/{endpoint}"
+        if options is not None and "callback" in options:
+            raise DeepgramError("Callback cannot be provided as an option to a synchronous transcription. Use `transcribe_url_callback` instead.")
+        if is_url_source(source):
+            body = source
+        else:
+            raise DeepgramError("Unknown transcription source type")
+        return await self.post(url, options, json=body)
+        
+    async def transcribe_url_callback( self, source: UrlSource, callback:str, options: PrerecordedOptions = None, endpoint: str="v1/listen") -> AsyncPrerecordedResponse:
+        """
+        Transcribes audio from a URL source and sends the result to a callback URL.
+
+        Args:
+            source (UrlSource): The URL source of the audio to transcribe.
+            callback (str): The callback URL where the transcription results will be sent.
+            options (PrerecordedOptions): Additional options for the transcription (default is None).
+            endpoint (str): The API endpoint for the transcription (default is "v1/listen").
+
+        Returns:
+            AsyncPrerecordedResponse: An object containing the request_id or an error message.
+
+        Raises:
+            DeepgramError: If the "callback" option is provided for a synchronous transcription.
```

Comment:
This is not relevant in the context of the `transcribe_url_callback()` method, right? I see further down in this method you're raising `raise DeepgramError("Unknown transcription source type")` for a non-URL source. Maybe that should be mentioned instead in this line about why a `DeepgramError` would be raised?

Also applied to the same comment in `transcribe_file_callback()`.